src/riak_kv_crdt.erl

```diff
@@ -290,27 +318,29 @@ counter_op(N) ->
 %% and risk precondition errors and unexpected behaviour.
 %%
 %% @see split_ops/1
--spec update_crdt(orddict:orddict(), riak_dt:actor(), crdt_op() | non_neg_integer()) ->
-                         orddict:orddict() | precondition_error().
-update_crdt(Dict, Actor, Amt) when is_integer(Amt) ->
+-spec update_crdt(orddict:orddict(), riak_dt:actor(),
+                  riak_object:bucket(), crdt_op() |
+                  non_neg_integer()) -> orddict:orddict() | precondition_error().
+update_crdt(Dict, Actor, Bucket, Amt) when is_integer(Amt) ->
     %% Handle legacy 1.4 counter operation, upgrade to current OP
     CounterOp = counter_op(Amt),
     Op = ?CRDT_OP{mod=?V1_COUNTER_TYPE, op=CounterOp},
-    update_crdt(Dict, Actor, Op);
-update_crdt(Dict, Actor, ?CRDT_OP{mod=Mod, op=Op, ctx=undefined}) ->
-    {Meta, Record, Value} = fetch_with_default(Mod, Dict),
+    update_crdt(Dict, Actor, Bucket, Op);
+update_crdt(Dict, Actor, Bucket, ?CRDT_OP{mod=Mod, op=Op, ctx=undefined}) ->
+    {Meta, Record, Value} = fetch_with_default(Mod, Dict, Bucket),
     case Mod:update(Op, Actor, Value) of
         {ok, NewVal} ->
             orddict:store(Mod, {Meta, Record?CRDT{value=NewVal}}, Dict);
         {error, _}=E -> E
     end;
-update_crdt(Dict, Actor, ?CRDT_OP{mod=Mod, op=Ops, ctx=OpCtx}) when Mod==?MAP_TYPE;
-                                                                    Mod==?SET_TYPE->
+update_crdt(Dict, Actor, Bucket, ?CRDT_OP{mod=Mod, op=Ops, ctx=OpCtx})
```

Comment:
IN fact the more I look, the more I can't figure out why the two clauses, the second seems to just duplicate `fetch_with_default`. I know this is legacy code, but maybe we should take this chance to refactor?